### ISSUE TITLE
采用dense_vector_sequence数据类型训练数据时报错

### ISSUE THREAD

**Issue by hawk199** [PaddlePaddle/Paddle#10064]
根据数字识别搭建了多层感知器网络，启动训练报错，不知啥原因
paddle.init(use_gpu=False, trainer_count=1)

    # define network topology
    match_analysis = paddle.layer.data(
        name='match', type=paddle.data_type.dense_vector_sequence(4))
    label = paddle.layer.data(
        name='label', type=paddle.data_type.integer_value(3))

    # Here we can build the prediction network in different ways. Please
    # choose one by uncomment corresponding line.
    # predict = softmax_regression(images)
    predict = multilayer_perceptron(match_analysis)

def multilayer_perceptron(match_analysis):
    # The first fully-connected layer
    hidden1 = paddle.layer.fc(input=match_analysis, size=128, act=paddle.activation.Relu())
    # The second fully-connected layer and the according activation function
    hidden2 = paddle.layer.fc(
        input=hidden1, size=64, act=paddle.activation.Relu())
    # The thrid fully-connected layer, note that the hidden size should be 10,
    # which is the number of unique digits
    predict = paddle.layer.fc(
        input=hidden2, size=10, act=paddle.activation.Softmax())
    return predict

数据为如下格式：
[[2.58, 3.25, 2.9, 2.0], [2.53, 3.25, 2.95, 27.0], [2.51, 3.25, 3.0, 47.0], [2.49, 3.25, 3.0, 55.0], [2.47, 3.3, 3.0, 120.0], [2.42, 3.3, 3.05, 152.0], [2.44, 3.3, 3.05, 212.0], [2.47, 3.3, 3.0, 243.0], [2.44, 3.3, 3.1, 283.0], [2.44, 3.2, 3.1, 910.0], [2.44, 3.15, 3.1, 1058.0], [2.42, 3.15, 3.15, 1289.0], [2.42, 3.05, 2.75, 1857.0], [2.39, 3.05, 2.8, 5735.0], [2.39, 3.2, 2.7, 6625.0], [2.35, 3.2, 2.75, 6626.0], [2.39, 3.2, 2.7, 6627.0]]

报错如下：
I0419 12:26:53.088719     8 Util.cpp:166] commandline:  --use_gpu=False --trainer_count=1
W0419 12:26:53.088758     8 CpuId.h:112] PaddlePaddle wasn't compiled to use avx instructions, but these are available o
n your machine and could speed up CPU computations via CMAKE .. -DWITH_AVX=ON
I0419 12:26:53.095015     8 GradientMachine.cpp:94] Initing parameters..
I0419 12:26:53.096026     8 GradientMachine.cpp:101] Init parameters done.
1416
F0419 12:26:53.189138     8 Matrix.cpp:3804] Check failed: label.getSize() == numSamples (4 vs. 135)
*** Check failure stack trace: ***
    @     0x7f1bfd49749d  google::LogMessage::Fail()
    @     0x7f1bfd49af4c  google::LogMessage::SendToLog()
    @     0x7f1bfd496fc3  google::LogMessage::Flush()
    @     0x7f1bfd49c45e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f1bfd3ce971  paddle::CpuMatrix::oneHotCrossEntropy()
    @     0x7f1bfd1b4965  paddle::CostLayer::forward()
    @     0x7f1bfd1909f9  paddle::NeuralNetwork::forward()
    @     0x7f1bfd168d83  paddle::GradientMachine::forwardBackward()
    @     0x7f1bfd473999  GradientMachine::forwardBackward()
    @     0x7f1bfd1132ac  _wrap_GradientMachine_forwardBackward
    @           0x4c30ce  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4c1e6f  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4c16e7  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4c16e7  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4c1e6f  PyEval_EvalFrameEx
    @           0x4b9ab6  PyEval_EvalCodeEx
    @           0x4eb30f  (unknown)
    @           0x4e5422  PyRun_FileExFlags
    @           0x4e3cd6  PyRun_SimpleFileExFlags
    @           0x493ae2  Py_Main
    @     0x7f1c16873830  __libc_start_main
    @           0x4933e9  _start
    @              (nil)  (unknown)
Aborted

**Comment by jacquesqiao** [comment]
cost的配置呢？

**Comment by hawk199** [comment]
@jacquesqiao  
paddle.init(use_gpu=False, trainer_count=1)

    # define network topology
    match_analysis = paddle.layer.data(
        name='match', type=paddle.data_type.dense_vector_sequence(4))
    label = paddle.layer.data(
        name='label', type=paddle.data_type.integer_value(3))

    # Here we can build the prediction network in different ways. Please
    # choose one by uncomment corresponding line.
    # predict = softmax_regression(images)
    predict = multilayer_perceptron(match_analysis)
    #predict = convolutional_neural_network(images)

    cost = paddle.layer.classification_cost(input=predict, label=label)

    parameters = paddle.parameters.create(cost)

    optimizer = paddle.optimizer.Momentum(
        learning_rate=0.1 / 128.0,
        momentum=0.9,
        regularization=paddle.optimizer.L2Regularization(rate=0.0005 * 128))

    trainer = paddle.trainer.SGD(
        cost=cost, parameters=parameters, update_equation=optimizer)

**Comment by shanyi15** [comment]
您好，此issue在近一个月内暂无更新，我们将于今天内关闭。若在关闭后您仍需跟进提问，可重新开启此问题，我们将在24小时内回复您。因关闭带来的不便我们深表歉意，请您谅解~感谢您对PaddlePaddle的支持! 
 Hello, this issue has not been updated in the past month. We will close it today for the sake of other user‘s experience. If you still need to follow up on this question after closing, please feel free to reopen it. In that case, we will get back to you within 24 hours. We apologize for the inconvenience caused by the closure and thank you so much for your support of PaddlePaddle Group!
